Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,480 @@
+            newActModel.Active = true;
+            return newActModel;
+        }
+
+        /// <summary>
+        /// This method is used to check if the app parameter is already added in the list if not then it will add the parameter
+        /// </summary>
+        /// <param name="aPIModel"></param>
+        /// <param name="act"></param>
+        /// <param name="actApiModel"></param>
+        private void AddAppParameters(ApplicationAPIModel aPIModel, Act act, ActWebAPIModel actApiModel, bool parameterizeRequestBody, bool isModelExists)
+        {
+            try
+            {
+                ObservableList<AppModelParameter> lstParameters = null;
+                ObservableList<ActInputValue> actInputs = ((ActWebAPIBase)act).DynamicElements;
+                if (actInputs == null || actInputs.Count == 0)
+                {
+                    string requestBody = string.Empty;
+                    if (aPIModel.RequestBodyType == ApplicationAPIUtils.eRequestBodyType.TemplateFile)
+                    {
+                        string fileUri = WorkSpace.Instance.SolutionRepository.ConvertSolutionRelativePath(aPIModel.TemplateFileNameFileBrowser);
+                        if (File.Exists(fileUri))
+                        {
+                            requestBody = System.IO.File.ReadAllText(fileUri);
+                        }
+                    }
+                    else
+                    {
+                        if (string.IsNullOrEmpty(aPIModel.RequestBody))
+                        {
+                            SetPropertyValue(aPIModel, nameof(ActWebAPIBase.Fields.RequestBody), nameof(ApplicationAPIModel.RequestBody), act);
+                            requestBody = Convert.ToString(aPIModel.RequestBody);
+                        }
+                        else
+                        {
+                            requestBody = Convert.ToString(aPIModel.RequestBody);
+                        }
+                    }
+
+                    ObservableList<ApplicationAPIModel> applicationAPIModels = new ObservableList<ApplicationAPIModel>();
+                    if (!string.IsNullOrEmpty(requestBody))
+                    {
+                        if (aPIModel.RequestBody.StartsWith("{"))
+                        {
+                            JSONTemplateParser jsonTemplate = new JSONTemplateParser();
+                            jsonTemplate.ParseDocumentWithJsonContent(requestBody, applicationAPIModels);
+                        }
+                        else if (aPIModel.RequestBody.StartsWith("<"))
+                        {
+                            XMLTemplateParser parser = new XMLTemplateParser();
+                            parser.ParseDocumentWithXMLContent(requestBody, applicationAPIModels);
+                        }
+
+                        if (!isModelExists && applicationAPIModels[0].AppModelParameters != null && applicationAPIModels[0].AppModelParameters.Count > 0)
```

Comment:
why to do all parameterization process and only then to check if the API model exists or not? check it on start and don't do anything related to parametrize if it's exist

Reply:
This is done since we should add parameters to action even if we do not want to add it to apimodel, this method is adding parameters to both apimodel and action in 1 go